admin/services/data-access/pupil-attendance.data.service.js

```diff
@@ -70,5 +70,13 @@ pupilAttendanceDataService.findByPupilIds = async (ids) => {
   const sql = [select, whereClause].join(' ')
   return sqlService.query(sql, params)
 }
+/**
+ * Returns pupil attendance based on pupil id
+ * @param pupilId
+ * @return {Object}
+ */
+pupilAttendanceDataService.findOneByPupilId = async (pupilId) => {
+  return sqlService.findOneByPupilId(table, pupilId)
```

Comment:
???  the sqlService is a generic data layer utility.  It should not have table specific methods

Reply:
@GuyHarwood was wrong in the first place, updated